src/Microsoft.CodeQuality.Analyzers/UnitTests/QualityGuidelines/ValidateArgumentsOfPublicMethodsTests.cs

```diff
@@ -1932,6 +1932,97 @@ private static void NoValidate(C c)
             GetCSharpResultAt(14, 13, "void C.M1(C c1, C c2)", "c2"));
         }
 
+        [Fact, WorkItem(2525, "https://github.com/dotnet/roslyn-analyzers/issues/2525")]
+        public void ValidatedNotNullAttributeInInvokedMethod_EditorConfig_NoInterproceduralAnalysis_NoDiagnostic_02()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+[AttributeUsage(AttributeTargets.Parameter)]
+internal sealed class ValidatedNotNullAttribute : Attribute { }
+
+internal static class Param
+{
+    public static void RequireNotNull([ValidatedNotNull] object value)
+    {
+    }
+}
+
+public class DataThing
+{
+    public IList<object> Items { get; }
+}
+
+public static class Issue2578Test
+{
+    public static void DoSomething(DataThing input)
+    {
+        Param.RequireNotNull(input);
+
+        // This line still generates a CA1062 error.
```

Comment:
>still generates [](start = 21, length = 15)

Now that it's fixed, it should say "shouldn't generate", right?

Reply:
Yes, but I am normally very lazy to edit provided minimal repros :). I hope the work item link above makes it easy to reason about it, though I will make sure to strip the comments from repro code going forward. Thanks!